tests/driver_ws281x/main.c

```diff
@@ -0,0 +1,102 @@
+/*
+ * Copyright 2019 Marian Buschsieweke
+ *
+ * This file is subject to the terms and conditions of the GNU Lesser
+ * General Public License v2.1. See the file LICENSE in the top level
+ * directory for more details.
+ */
+
+/**
+ * @ingroup tests
+ * @{
+ *
+ * @file
+ * @brief   Test application for the NeoPixel RGB LED driver
+ *
+ * @author  Marian Buschsieweke <marian.buschsieweke@ovgu.de>
+ *
+ * @}
+ */
+
+#include <stdio.h>
+
+#include "ws281x.h"
+#include "ws281x_params.h"
+#include "xtimer.h"
+
+static const color_rgb_t rainbow[] = {
+    {.r = 0x94, .g = 0x00, .b = 0xd3},
+    {.r = 0x4b, .g = 0x00, .b = 0x82},
+    {.r = 0x00, .g = 0x00, .b = 0xff},
+    {.r = 0x00, .g = 0xff, .b = 0x00},
+    {.r = 0xff, .g = 0xff, .b = 0x00},
+    {.r = 0xff, .g = 0x7f, .b = 0xd3},
+    {.r = 0xff, .g = 0x00, .b = 0x00},
+};
+
+#define RAINBOW_LEN     (sizeof(rainbow) / sizeof(rainbow[0]))
+
+int main(void)
+{
+    ws281x_t dev;
+    int retval;
+
+    puts(
+        "NeoPixel test application\n"
+        "=========================\n"
+        "\n"
+        "If you see an animated rainbow, the driver works as expected.\n"
+        "If the LEDs are flickering, check if the power supply is sufficient\n"
+        "(at least 4V). Also: The logic level has to be at least 0.7 * VDD,\n"
+        "so 3.3V logic with a 5V power supply is out of spec, but might work\n"
+        "OK.\n"
+    );
+
+    if (0 != (retval = ws281x_init(&dev, &ws281x_params[0]))) {
+        printf("Initialization failed with error code %d\n", retval);
+        return retval;
+    }
+
+    while (1) {
+        unsigned offset = 0;
+        puts("Animation: Moving rainbow...");
+        for (unsigned i = 0; i < 100; i++) {
+            for (uint16_t j = 0; j < dev.params.numof; j++) {
+                ws281x_set(&dev, j, rainbow[(j + offset) % RAINBOW_LEN]);
+            }
+            offset++;
+            ws281x_write(&dev);
+            xtimer_usleep(100 * US_PER_MS);
```

Comment:
I think `xtimer_periodic_wakeup()` would be better suited here.

Reply:
Done